src/screens/VaultScreen.js

```diff
@@ -13,11 +14,305 @@ export default class VaultScreen extends React.Component {
     ),
   };
 
+  async componentDidMount() {
```

Comment:
Just a nit for style--you should put constructor first and then lifecycle methods in order afterwards: https://github.com/airbnb/javascript/tree/master/react#ordering